src/scales/scale.time.js

```diff
@@ -175,8 +175,20 @@ module.exports = function(Chart) {
 		var ticks = [];
 		if (options.maxTicks) {
 			var stepSize = options.stepSize;
-			ticks.push(options.min !== undefined ? options.min : niceRange.min);
-			var cur = moment(niceRange.min);
+			var startTick = options.min !== undefined ? options.min : niceRange.min;
+			var majorUnitStart = startTick;
+			if (options.majorUnit) {
+				majorUnitStart = moment(startTick).add(1, options.majorUnit).startOf(options.majorUnit);
+			}
+			var startRange = majorUnitStart.valueOf() - startTick;
+			var startFraction = startRange % (interval[options.unit].size * stepSize);
+			var alignedTick = startTick;
+			ticks.push(startTick);
+			if (startTick !== alignedTick + startFraction && options.majorUnit && !options.timeOpts.round && !options.timeOpts.isoWeekday) {
```

Comment:
@hurskiy-andriy I'm not following your example. I don't understand why `majorUnitStart` will be equal to "00:00". Wouldn't it be equal to "21:00"?

`majorUnitStart = moment(startTick).add(1, options.majorUnit).startOf(options.majorUnit);`
majorUnitStart = (20.12 + 1hr).startOf(hour) = 21.12.startOf(hour) = 21:00